### DISCUSSION TITLE
resolution ratio

### DISCUSSION THREAD

**Discussion by jiangwu007** [israel-dryer/ttkbootstrap#336] [Q&A] [answered]
Under different computers, the resolution is different Each time you want to select a resolution, close the program, and then open the program. Can you realize a function to automatically switch the resolution?
![fen](https://user-images.githubusercontent.com/21976698/185772520-6124a4c8-5d70-4d1c-aca7-437369df6f17.png)

**Comment by israel-dryer** [comment, marked as answer]
This is possible, but it will take some changes to the architecture of the library to achieve. This is one the roadmap for version 2.0.

There is a [scaling function in tcl/tk](https://tcl.tk/man/tcl8.6/TkCmd/tk.htm#M10) that can be used. However, the images that I'm using for the styles are created with the scaling at the time the styles is generated, so a change to scaling would not affect this image-based styles currently.

However, this does sounds like a reasonable thing to do, so I will look at how this can be implemented in version 2.0 of ttkbootstrap which I'm working on now.

To document this for my own notes later... I need to create and generate a <<ResolutionChanged>> event when the resolution changes. Add an event handler (on_resolution_changed) that will erase the style assets and registry. Then, generate a '<<StyleChanged>>' event that will automatically cause the styles for all widgets to rebuild with the new scaling. I'll need to test whether or not the widgets resize their own geometry as this is an open question in the current documentation (perhaps widget specific?).

**Reply by jiangwu007** [reply]
Great!
